var/spack/repos/builtin/packages/py-datasets/package.py

```diff
@@ -18,30 +18,44 @@ class PyDatasets(PythonPackage):
 
-    depends_on("python@3.6:", type=("build", "run"))
-    depends_on("py-setuptools", type="build")
-    depends_on("py-numpy@1.17:", type=("build", "run"))
-    depends_on("py-pyarrow@1:3+parquet", type=("build", "run"), when="@:1.8.0")
-    depends_on("py-pyarrow@6:+parquet", type=("build", "run"), when="@2.8.0:")
-    depends_on("py-dill@:0.3.6", type=("build", "run"))
-    depends_on("py-pandas", type=("build", "run"))
-    depends_on("py-requests@2.19:", type=("build", "run"))
-    depends_on("py-tqdm@4.27:4.49", type=("build", "run"), when="@:1.8.0")
-    depends_on("py-tqdm@4.62.1:", type=("build", "run"), when="@2.8.0:")
-    depends_on("py-xxhash", type=("build", "run"))
-    depends_on("py-multiprocess", type=("build", "run"))
-    depends_on("py-importlib-metadata", when="^python@:3.7", type=("build", "run"))
-    depends_on("py-huggingface-hub@:0.0", type=("build", "run"), when="@:1.8.0")
-    depends_on("py-huggingface-hub@0.2:0", type=("build", "run"), when="@2.8.0:")
-    depends_on("py-packaging", type=("build", "run"))
-    depends_on("py-aiohttp", type=("build", "run"), when="@2.8.0:")
-    depends_on("py-responses@:0.18", type=("build", "run"), when="@2.8.0:")
-    depends_on("py-pyyaml@5.1:", type=("build", "run"), when="@2.8.0:")
-
-    with when("@:1.8.0"):
-        depends_on("py-fsspec", type=("build", "run"), when="^python@3.8:")
-        depends_on("py-fsspec@:0.8.0", type=("build", "run"), when="^python@:3.7")
-    depends_on("py-fsspec@2021.11.1:+http", type=("build", "run"), when="@2.8.0:")
+    with default_args(type="build"):
+        depends_on("py-setuptools")
+
+    with default_args(type=("build", "run")):
+        depends_on("python@3.6:")
+        depends_on("py-numpy@1.17:")
+        depends_on("py-pandas")
+        depends_on("py-requests@2.19:")
+        depends_on("py-xxhash")
+        depends_on("py-multiprocess")
+        depends_on("py-packaging")
+        with when("@:1.8.0"):
+            depends_on("py-dill@:0.3.6")
+            depends_on("py-fsspec", when="^python@3.8:")
+            depends_on("py-fsspec@:0.8.0", when="^python@:3.7")
+            depends_on("py-huggingface-hub@:0.0")
+            depends_on("py-importlib-metadata", when="^python@:3.7")
+            depends_on("py-pyarrow@1:3+parquet")
+            depends_on("py-tqdm@4.27:4.49", when="@:1.8.0")
+        with when("@2.8.0:"):
+            depends_on("py-aiohttp")
+            depends_on("py-pyyaml@5.1:")
+        with when("@2.8.0"):
+            depends_on("py-dill@:0.3.6")
+            depends_on("py-fsspec@2021.11.1:+http")
+            depends_on("py-huggingface-hub@0.2:0")
+            depends_on("py-pyarrow@6:+parquet")
+            depends_on("py-responses@:0.18")
+            depends_on("py-tqdm@4.62.1:")
+        with when("@2.20.0:"):
+            depends_on("py-filelock")
+            depends_on("py-dill@0.3.0:0.3.8")  # temporary upper bound
+            depends_on("py-fsspec@2023.1.0:2024.5.0")
```

Comment:
My impression is that

```
    # minimum 2023.1.0 to support protocol=kwargs in fsspec's `open`, `get_fs_token_paths`, etc.: see https://github.com/fsspec/filesystem_spec/pull/1143
    "fsspec[http]>=2023.1.0,<=2024.5.0",
```

means the dependency needs to include the enabled variant:

```suggestion
            depends_on("py-fsspec@2023.1.0:2024.5.0+http")
```

Reply:
Yes removed it by mistake, thank you!